project_tracker/cluster.py

```diff
@@ -263,9 +268,11 @@ def fit_bounding_boxes(self, tracked_detected_objects):
             bounding_box_marker.color.g = 255.#self.rgb_list[cluster_idx][1]/255.
             bounding_box_marker.color.b = 255.#self.rgb_list[cluster_idx][2]/255.
             # size -> 2 times the max_point from centre
-            bounding_box_marker.scale.x = d_o.dimensions.x
-            bounding_box_marker.scale.y = d_o.dimensions.y
-            bounding_box_marker.scale.z = d_o.dimensions.z
+
+            # Flipping bounding box fit
+            bounding_box_marker.scale.x = d_o.dimensions.y
```

Comment:
this is a good find but do we know why we're mapping detected object's y coordinate to bounding box's x?

Reply:
Good stuff! I think it's worth digging deeper into it as Amir said, to make sure we aren't going to make an incorrect assumption about coordinate frames somewhere else.

Reply:
removed this logic and fixed the root problem which happens when the detected object is being created. marking as resolved